src/ResourceManager/CognitiveServices/CognitiveServices.Test/ScenarioTests/CognitiveServicesAccountTests.ps1

```diff
@@ -0,0 +1,391 @@
+        $skuNames = $skus | Select-Object -ExpandProperty Name
+        
+        $expectedSkus = "F0", "S1", "S2", "S3", "S4"
+        Assert-AreEqualArray $expectedSkus $skuNames
+
+        Retry-IfException { Remove-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname; }
+    }
+    finally
+    {
+        # Cleanup
+        Clean-ResourceGroup $rgname
+    }
+}
+
+<#
+.SYNOPSIS
+Test Get-AzureRmCognitiveServicesAccount | Get-AzureRmCognitiveServicesAccountKey
+#>
+function Test-PipingGetAccountToGetKey
+{
+    # Setup
+    $rgname = Get-CognitiveServicesManagementTestResourceName;
+
+    try
+    {
+        # Test
+        $accountname = 'csa' + $rgname;
+        $skuname = 'S2';
+        $accounttype = 'TextAnalytics';
+        $loc = 'West US';
+
+        New-AzureRmResourceGroup -Name $rgname -Location $loc;
+        New-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname -Type $accounttype -SkuName $skuname -Location $loc;
+
+        $keys = Get-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname | Get-AzureRmCognitiveServicesAccountKey;
+        Assert-AreNotEqual $keys.Key1 $keys.Key2;
+
+        Retry-IfException { Remove-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname; }
+    }
+    finally
+    {
+        # Cleanup
+        Clean-ResourceGroup $rgname
+    }
+}
+
+<#
+.SYNOPSIS
+Test Get-AzureRmCognitiveServicesAccount | Set-AzureRmCognitiveServicesAccount
+#>
+function Test-PipingToSetAzureAccount
+{
+ # Setup
+    $rgname = Get-CognitiveServicesManagementTestResourceName
+
+    try
+    {
+        # Test
+        $accountname = 'csa' + $rgname;
+        $skuname = 'S2';
+        $accounttype = 'TextAnalytics';
+        $loc = 'West US';
+
+        New-AzureRmResourceGroup -Name $rgname -Location $loc;
+        New-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname -Type $accounttype -SkuName $skuname -Location $loc;
+
+        $account = Get-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname;
+		$account | Set-AzureRmCognitiveServicesAccount -SkuName S3;
+		
+        $updatedAccount = Get-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname;
+        Assert-AreEqual 'S3' $updatedAccount.Sku.Name;
+
+        Retry-IfException { Remove-AzureRmCognitiveServicesAccount -ResourceGroupName $rgname -Name $accountname; }
+    }
+    finally
+    {
+        # Cleanup
+        Clean-ResourceGroup $rgname
+    }
+}
+
+<#
+.SYNOPSIS
+Test Get-AzureRmCognitiveServicesAccount | Get-AzureRmCognitiveServicesAccountSkus
+#>
+function Test-PipingToGetAccountSkus
+{
+ # Setup
+    $rgname = Get-CognitiveServicesManagementTestResourceName
+
+    try
+    {
+        # Test
+        $accountname = 'csa' + $rgname;
+        $skuname = 'S2';
+        $accounttype = 'TextAnalytics';
+        $loc = 'West US';
```

Comment:
line breakDo not hard code locations.  Use the common code to retrieve a valid location.  This can prevent live tests from running in some environments